jwst/cube_build/src/cube_match_internal.c

```diff
@@ -220,45 +205,66 @@ PyArrayObject * ensure_array(PyObject *obj, int *is_copy) {
     }
 }
 
+PyArrayObject * ensure_array_int(PyObject *obj, int *is_copy) {
+  if (PyArray_CheckExact(obj) &&
+      PyArray_IS_C_CONTIGUOUS((PyArrayObject *) obj) &&
+      PyArray_TYPE((PyArrayObject *) obj) == NPY_UINT32) {
+    *is_copy = 0;
+    return (PyArrayObject *) obj;
+  } else {
+    *is_copy = 1;
+    return (PyArrayObject *) PyArray_FromAny(obj, PyArray_DescrFromType(NPY_UINT32), 0, 0,
+					     NPY_ARRAY_CARRAY | NPY_ARRAY_FORCECAST, NULL
+					     );
+  }
+}
+
 
 static PyObject *cube_wrapper_internal(PyObject *module, PyObject *args) {
 
-  PyObject *result = NULL, *a1o, *a2o, *a3o, *a4o, *lam1o, *lam2o, *lam3o, *lam4o,
+  PyObject  *a1o, *a2o, *a3o, *a4o, *lam1o, *lam2o, *lam3o, *lam4o,
     *fluxo, *erro, *acoordo, *zcoordo;
 
+  PyObject *spaxel_fluxo, *spaxel_weighto, *spaxel_varo, *spaxel_ifluxo;  
   double crval_along, cdelt_along, crval3, cdelt3;
   int  nz, na, npt, naxis1, naxis2, ncube;
   int ss;
   int instrument;
 
-  double *spaxel_flux=NULL, *spaxel_weight=NULL, *spaxel_var=NULL;
-  double *spaxel_iflux=NULL;
-
   int free_a1=0, free_a2=0, free_a3=0, free_a4 =0, free_lam1=0, free_lam2 =0, free_lam3=0, free_lam4=0;
   int free_acoord=0, free_zcoord=0, status=0;
   int free_flux=0, free_err=0;
-
+  int free_spaxel_flux=0, free_spaxel_weight=0, free_spaxel_var=0, free_spaxel_iflux=0;
+  
   int n1, n2;
-  PyArrayObject *a1=NULL, *a2=NULL, *a3=NULL, *a4=NULL, *lam1=NULL, *lam2=NULL;
-  PyArrayObject *lam3=NULL, *lam4=NULL, *flux=NULL, *err=NULL, *acoord=NULL, *zcoord=NULL;
-
-  PyArrayObject *spaxel_flux_arr=NULL, *spaxel_weight_arr=NULL, *spaxel_var_arr=NULL;
-  PyArrayObject *spaxel_iflux_arr=NULL;
-  npy_intp npy_ncube = 0;
-
-  if (!PyArg_ParseTuple(args, "iiiddddOOOOOOOOOOiOO:cube_wrapper_internal",
+  PyArrayObject *a1, *a2, *a3, *a4, *lam1, *lam2;
+  PyArrayObject *lam3, *lam4, *flux, *err, *acoord, *zcoord;
+  PyArrayObject *spaxel_flux, *spaxel_weight, *spaxel_var, *spaxel_iflux;
+  
+  const int max_size_error = 80;
```

Comment:
Now `const int max_size_error = 80;` is not used anywhere. The correct way of doing this in C is to use macros:
```C
#define MAX_ERROR_LENGTH 80
```
somewhere at the top of the file and then inside the function do something like:
```C
char error[MAX_ERROR_LENGTH] = "Hello World";
```

Reply:
But why limit error message length at all? Is that important?